sw/device/tests/sram_ctrl_sleep_sram_ret_contents_test.c

```diff
@@ -100,22 +135,24 @@ bool test_main(void) {
           kDifToggleEnabled));
 
       CHECK_DIF_OK(dif_aon_timer_wakeup_stop(&aon_timer));
-      aon_timer_testutils_watchdog_config(&aon_timer, 0xffffffff, 10, false);
 
       // Enter low power mode.
       LOG_INFO("Issue WFI to enter sleep");
+      aon_timer_testutils_watchdog_config(&aon_timer, UINT32_MAX, 20, false);
       wait_for_interrupt();
     } else {
       LOG_INFO("Watchdog reset");
       // Check that the previously written retention SRAM data can still
       // be read successfully.
-      retention_sram_check(false);
+      retention_sram_check(
+          (check_config_t){.do_write = false, .is_equal = false});
```

Comment:
Should the key really reset after watchdog reset? @weicaiyang @msfschaffner

Reply:
Yes, since the SRAM logic is reset when coming back from deep sleep.

So, even though the retention SRAM is not powered off (since it is located in the always-on domain), the key register will be reset back to the default.

Now, preserving data across deep sleep is still possible by just using the default key that is present after reset (i.e., not requesting a scrambling key at the beginning).

This makes me think that it may be good to add another test iteration to check this last point.

That way, we should cover the combinations that we care about:
1) light sleep exit + ephemeral key: data needs to be preserved
2) deep sleep exit + ephemeral key: data is NOT preserved as key is reset
3) deep sleep exist + default key: data needs to be preserved

@weicaiyang WDYT?

Reply:
Thanks @msfschaffner